crypto/evp/exchange.c

```diff
@@ -20,6 +20,9 @@ static EVP_KEYEXCH *evp_keyexch_new(OSSL_PROVIDER *prov)
 {
     EVP_KEYEXCH *exchange = OPENSSL_zalloc(sizeof(EVP_KEYEXCH));
 
+    if (exchange == NULL)
+        return NULL;
```

Comment:
In [line 43/46](https://github.com/openssl/openssl/pull/10526/commits/4c0c53b009d6a3e7a44b43d8238698faf8ad674d#diff-7f75adab80125f8e46d5e500b71fff74R46), you raise an `ERR_R_MALLOC_FAILURE` when heap allocation fails. Shouldn't this be done consistently for all malloc failures? (several locations)

Reply:
_Note: if you add new error codes, take a look at issue #10522 before running `make update`._